### PR TITLE
Add plugin development notes in a readme.

### PR DESCRIPTION
**Pull request by mashhurs**
Adds plugin development notes to the readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,4 +7,74 @@ It can be used in a Logstash pipeline to perform the transformations that are ap
...
+
+## Development
+### Prerequisites
+- Clone the [plugin repo](https://github.com/elastic/logstash-filter-elastic_integration)
+- Check out the branch you would like to build, such as `8.x` or `9.x/main`, etc...
+  - Note that, `8.series` branches require JDK17 and `9.series`/`main` requires JDK21.
+- Set the Logstash source path in the `gradle.properties` file. Open the file and update the `LOGSTASH_PATH`.
+- Make sure Logstash is compiled.
+- Point to the Elasticsearch branch/version to build the plugin with. Open the `gradle.properties` file and update `ELASTICSEARCH_TREEISH`.
+- Point to the Elasticsearch branch/version to build the plugin with. Open the `gradle.properties` file and update `ELASTICSEARCH_TREEISH`.
```
Comment (donoghuc) [inline README.md:20]:
```suggestion
```
duplicate?

Comment (mashhurs) [reply README.md:20]: Yup, indeed duplicate, thank you for catching.

File: README.md [hunk 2]
```diff
@@ -7,4 +7,74 @@ It can be used in a Logstash pipeline to perform the transformations that are ap
...
+
+### Build
+We use Gradle tool to build this plugin. Gradle seeks Logstash core jars from the `LOGSTASH_PATH` path, downloads the Elasticsearch version defined with `ELASTICSEARCH_TREEISH` and build the plugin.
```
Comment (donoghuc) [inline README.md:22]:
```suggestion
We use Gradle tool to build this plugin. Gradle is configured to load the Logstash core jars from the `LOGSTASH_PATH` path, download the Elasticsearch version defined with `ELASTICSEARCH_TREEISH` and build the plugin.
```

**Review by donoghuc — APPROVED** [review]
Content looks great. One small readability suggestion: https://github.com/elastic/logstash-filter-elastic_integration/pull/209#discussion_r1890764368

**Review by donoghuc — APPROVED** [review]

**Comment by mashhurs** [comment]
testing backport

**Comment by mashhurs** [comment]
@logstashmachine backport 8.x

**Comment by mashhurs** [comment]
@logstashmachine backport 8.x

**Comment by mashhurs** [comment]
@logstashmachine backport 8.17

**Comment by mashhurs** [comment]
@logstashmachine backport 8.16

**Pull request merged**